### PR TITLE
Use Value dim shape for Attention compute_output_shape

### LINKED ISSUE #19257: Keras 3 Attention layer value tensor dimension
hi,

I found the  below would not return the proper size output in Keras 3 (but works fine in Keras 2)
Please help to fix it, 

Thanks.

```python
import keras
from keras import layers

i = layers.Input((8,4))
xq = layers.Conv1D(5,1)(i)
xk = layers.Conv1D(5,1)(i)
xv = layers.Conv1D(7,1)(i)
o = layers.Attention()([xq,xv,xk])

m = keras.Model(inputs=i, outputs=o)
m.summary()

Output as below
┏━━━━━━━━━━━━━━━━━━━━━┳━━━━━━━━━━━━━━━━━━━┳━━━━━━━━━━━━┳━━━━━━━━━━━━━━━━━━━┓
┃ Layer (type)        ┃ Output Shape      ┃    Param # ┃ Connected to      ┃
┡━━━━━━━━━━━━━━━━━━━━━╇━━━━━━━━━━━━━━━━━━━╇━━━━━━━━━━━━╇━━━━━━━━━━━━━━━━━━━┩
│ input_layer         │ (None, 8, 4)      │          0 │ -                 │
│ (InputLayer)        │                   │            │                   │
├─────────────────────┼───────────────────┼────────────┼───────────────────┤
│ conv1d (Conv1D)     │ (None, 8, 5)      │         25 │ input_layer[0][0] │
├─────────────────────┼───────────────────┼────────────┼───────────────────┤
│ conv1d_2 (Conv1D)   │ (None, 8, 7)      │         35 │ input_layer[0][0] │
├─────────────────────┼───────────────────┼────────────┼───────────────────┤
│ conv1d_1 (Conv1D)   │ (None, 8, 5)      │         25 │ input_layer[0][0] │
├─────────────────────┼───────────────────┼────────────┼───────────────────┤
│ attention           │ **(None, 8, 5)**      │          0 │ conv1d[0][0],     │
│ (Attention)         │                   │            │ conv1d_2[0][0],   │
│                     │                   │            │ conv1d_1[0][0]    │
└─────────────────────┴───────────────────┴────────────┴───────────────────┘
 Total params: 85 (340.00 B)
 Trainable params: 85 (340.00 B)
 Non-trainable params: 0 (0.00 B)
```

The Attention layer output shape should be (None, 8, 7), since **xv** is from **Conv1D** with 7 kernels.

Thie same code gives correct output from keras 2

```python
Model: "model"
__________________________________________________________________________________________________
 Layer (type)                   Output Shape         Param #     Connected to                     
==================================================================================================
 input_1 (InputLayer)           [(None, 8, 4)]       0           []                               
                                                                                                  
 conv1d (Conv1D)                (None, 8, 5)         25          ['input_1[0][0]']                
                                                                                                  
 conv1d_2 (Conv1D)              (None, 8, 7)         35          ['input_1[0][0]']                
                                                                                                  
 conv1d_1 (Conv1D)              (None, 8, 5)         25          ['input_1[0][0]']                
                                                                                                  
 attention (Attention)          **(None, 8, 7)**         0           ['conv1d[0][0]',                 
                                                                  'conv1d_2[0][0]',               
                                                                  'conv1d_1[0][0]']               
                                                                                                  
==================================================================================================
Total params: 85
Trainable params: 85
Non-trainable params: 0
```

### PR DESCRIPTION
**Pull request by sampathweb**
Fixes #19257 by using Value dim shape for `compute_output_shape` of Attention layer.

### CODE REVIEW

File: keras/layers/attention/attention_test.py [hunk 1]
```diff
@@ -342,3 +342,8 @@ def test_attention_compute_mask_with_different_input_shapes(self):
             computed_mask = layer.compute_mask(inputs=dummy_inputs, mask=mask)
             computed_mask = ops.convert_to_numpy(computed_mask)
             self.assertTrue(np.array_equal(computed_mask, valid_mask))
+
+    def test_attention_compute_output_shape(self):
+        layer = layers.Attention()
+        input_shape = [(2, 8, 7), (2, 8, 5), (2, 8, 7)]  # Shapes of Q, V, K
+        self.assertAllEqual(layer.compute_output_shape(input_shape) == (2, 8, 5))
```
Comment (fchollet) [inline keras/layers/attention/attention_test.py:349]: Please call the layer on an input and read its shape, to ensure match between actual shape and computed shape.

**Review by fchollet — COMMENTED** [review]
Thanks for the PR! Unit tests are failing, I think some golden values need to be updated.

File: keras/dtype_policies/dtype_policy.py [hunk 2]
```diff
@@ -173,9 +173,6 @@ def _parse_name(self, name):
             return "float16", "float32"
         elif name == "mixed_bfloat16":
             return "bfloat16", "float32"
-        elif name == "uint8":
```
Comment (sampathweb) [inline keras/dtype_policies/dtype_policy.py:176]: This is redundant.  Its addressed in the try block.

File: keras/layers/attention/attention_test.py (hunk 1 above)
Comment (sampathweb) [reply keras/layers/attention/attention_test.py]: Addressed the comment.
